tests/t0114_gateway_subdomains_test.go

```diff
@@ -13,49 +13,63 @@ import (
 
 func TestGatewaySubdomains(t *testing.T) {
 	// fixture := car.MustOpenUnixfsCar("t0114-gateway_subdomains")
-
-	CID_VAL := "hello"
+	// TODO: extract the CID correctly.
+	// TODO: how to extract CIDv1, CIDv0, and DIR_CID from the fixture?
 	CIDv1 := "bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am"
-	// CIDv0 := "QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN"
-	// // CIDv0to1 is necessary because raw-leaves are enabled by default during
-	// // "ipfs add" with CIDv1 and disabled with CIDv0
-	// CIDv0to1 := "bafybeiffndsajwhk3lwjewwdxqntmjm4b5wxaaanokonsggenkbw6slwk4"
-	// CIDv1_TOO_LONG := "bafkrgqhhyivzstcz3hhswshfjgy6ertgmnqeleynhwt4dlfsthi4hn7zgh4uvlsb5xncykzapi3ocd4lzogukir6ksdy6wzrnz6ohnv4aglcs"
-	DIR_CID := "bafybeiht6dtwk3les7vqm6ibpvz6qpohidvlshsfyr7l5mpysdw2vmbbhe" // ./testdirlisting
+	DIR_CID := "bafybeiht6dtwk3les7vqm6ibpvz6qpohidvlshsfyr7l5mpysdw2vmbbhe"
 
 	tests := []CTest{
 		{
-			Name: "request for 127.0.0.1/ipfs/{CID} stays on path",
-			Request: CRequest{
-				Url: fmt.Sprintf("ipfs/%s", CIDv1),
-			},
-			Response: CResponse{
-				StatusCode: 200,
-				Body:       Contains(CID_VAL),
-			},
-		},
-		{
-			Name: "request for localhost/ipfs/{CIDv1} returns HTTP 301 Moved Permanently",
+			Name: "request for {gateway}/ipfs/{CIDv1} returns HTTP 301 Moved Permanently",
 			Request: CRequest{
 				DoNotFollowRedirects: true,
-				Url:                  fmt.Sprintf("/ipfs/%s", CIDv1),
+				Url:                  fmt.Sprintf("%s/ipfs/%s", SubdomainGatewayUrl, CIDv1),
 			},
 			Response: CResponse{
 				StatusCode: 301,
 				Headers: map[string]interface{}{
-					"Location": Contains("http://%s.ipfs.localhost:8080", CIDv1),
+					"Location": Contains("%s://%s.ipfs.%s", SubdomainGatewayScheme, CIDv1, SubdomainGatewayHost),
 				},
 			},
 		},
+		{
+			Name: "request for {gateway}/ipfs/{CIDv1} returns HTTP 301 Moved Permanently (sugar)",
+			Request: Request().
+				URL("http://example.com/ipfs/%s", CIDv1).
+				DoNotFollowRedirects().
+				Request(),
+			Response: Expect().
+				Status(301).
+				Headers(
+					Header("Location").
+						// TODO: this works only because we use example.com in our tests.
+						// It should be:
+						// Contains("%s://%s.ipfs.%s", SubdomainGatewayScheme, CIDv1, SubdomainGatewayHost)
+						// I am trying to avoid this syntax.
+						// The other option is to force the tested gateway to use example.com.
+						Contains("http://%s.ipfs.example.com", CIDv1),
```

Comment:
I don't have a good solution for now, apart from wrapping with some function that signals `ThisIsASubdomainURL`.